### ISSUE TITLE
doc: add documentation for "debounce" in utils package

### ISSUE THREAD

**Issue by wsameer** [medly/medly-components#446]
## debounce
`utils > helpers > debounce`

## Context
Add documentation for `debounce` function in Medly Storybook.

**Must haves:**

- A detailed explanation of the function in layman language.
- A working example demonstrating the usage of the function.
- A sample unit test for testing the function. (good to have)

**Comment by Tajwinder** [comment]
Hi Sameer, I am working on it. so I assigning it to myself.
